crates/sui/src/unit_tests/upgrade_compatibility_tests.rs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::upgrade_compatibility::compare_packages;
+use insta::assert_debug_snapshot;
+use move_binary_format::CompiledModule;
+use std::path::PathBuf;
+use sui_move_build::BuildConfig;
+
+#[test]
+fn test_all_fail() {
+    let (pkg_v1, pkg_v2) = get_packages("all");
+
+    let result = compare_packages(pkg_v1, pkg_v2);
+    assert!(result.is_err());
+    let err = result.unwrap_err();
+
+    assert_debug_snapshot!(err);
```

Comment:
I don't think the debug snapshot has quite worked here -- it's still converting to a string, and then it's showing the debug of that, which is not what we want. If we want `Debug` to work, I think you need to drop the `Error` impl for the error type and all the associated `#[error(...)]` annotations, (I think that's fine to do, because your proposed error format is not likely to re-use that) or alternatively, switch to a regular `assert_snapshot!` and use the current string format.